### PR TITLE
Thread-safe Buffer.__items__; proper KeysView, ItemsView, ValuesView

### LINKED ISSUE #61: Dictionary views
The values() and items() methods of the various zict mappings return iterators.
The keys() methods return sometimes iterators, sometimes sets.
This violates the Python 3 MutableMapping API, which requires ValuesView, ItemsView, and KeysView respectively:

https://docs.python.org/3/library/collections.abc.html
https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects

### PR DESCRIPTION
**Pull request by crusaderky**
- Closes #61 
- Partially closes https://github.com/dask/distributed/issues/4424
- Fix RuntimeError when a thread is iterating on Buffer, e.g. `distributed.worker_state_machine.WorkerStateMachine.validate_state()`, while another is evicting/restoring a key

### CODE REVIEW

File: zict/func.py [hunk 1]
```diff
@@ -74,13 +74,6 @@ def __delitem__(self, key: KT) -> None:
     def keys(self) -> KeysView[KT]:
```
Comment (hendrikmakait) [inline zict/func.py:74]: nit: Can we also drop this? IIUC, this should be covered by the base implementation.

File: zict/buffer.py [hunk 2]
```diff
@@ -165,3 +178,25 @@ def flush(self) -> None:
 
     def close(self) -> None:
         close(self.fast, self.slow)
+
+
+class BufferItemsView(ItemsView[KT, VT]):
```
Comment (hendrikmakait) [inline zict/buffer.py:183]: Should we also override `__contains__` in `BufferItemsView` to avoid changing the LRU?

**Review by hendrikmakait — COMMENTED** [review]
I have two questions regarding the implementation and a few nits.

File: zict/func.py (hunk 1 above)
Comment (crusaderky) [reply zict/func.py:74]: yes

File: zict/lmdb.py [hunk 3]
```diff
@@ -110,3 +108,27 @@ def __len__(self) -> int:
...
     def close(self) -> None:
         self.db.close()
+
+
+class LMDBItemsView(ItemsView[str, bytes]):
+    _mapping: LMDB  # FIXME CPython implementation detail
+    __slots__ = ()
+
+    def __contains__(self, item: object) -> bool:
+        return any(item == v for v in self)
```
[the remark this answers is not in the document]
Comment (crusaderky) [reply zict/lmdb.py:118]: you're right!

File: zict/buffer.py (hunk 2 above)
Comment (hendrikmakait) [reply zict/buffer.py:183]: Thanks for the explanation; now it makes sense to me. I had assumed we wanted this to be a general property of the views.

>  If you're testing if your buffer contains a specific key/value _pair_, and that causes the value to be unevicted from slow, then I don't see why it should bypass the principle of temporal locality.

Sounds good to me!

**Comment by crusaderky** [comment]
@hendrikmakait all review comments have been addressed

**Review by hendrikmakait — APPROVED** [review]
Thanks, @crusaderky!

**Pull request merged**